packages/hardhat-core/src/internal/hardhat-network/provider/provider.ts

```diff
@@ -86,7 +87,10 @@ export class HardhatNetworkProvider
     private readonly _initialDate?: Date,
     private readonly _experimentalHardhatNetworkMessageTraceHooks: BoundExperimentalHardhatNetworkMessageTraceHook[] = [],
     private _forkConfig?: ForkConfig,
-    private readonly _forkCachePath?: string
+    private readonly _forkCachePath?: string,
+    private readonly _hardforkActivationBlocks?: {
```

Comment:
Consider creating a named type for this chainId mapping, like perhaps `HardforkActivationBlocksPerChain`. That will help resolve the bad naming of this field vs the type.